libs/vkd3d/command.c

```diff
@@ -6089,70 +6090,128 @@ static void d3d12_command_list_transition_image_layout(struct d3d12_command_list
+
+        out->vk_layout = d3d12_resource_pick_layout(src_resource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL);
+    }
+    else if (src->Type == D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT && dst->Type == D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX)
+    {
+        assert(d3d12_resource_is_texture(dst_resource));
+        assert(d3d12_resource_is_buffer(src_resource));
+
+        if (!(out->dst_format = vkd3d_format_from_d3d12_resource_desc(list->device,
+                      &dst_resource->desc, DXGI_FORMAT_UNKNOWN)))
+        {
+            WARN("Invalid format %#x.\n", dst_resource->desc.Format);
+            return false;
+        }
 
-        vk_layout = d3d12_resource_pick_layout(src_resource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL);
+        if (!(out->src_format = vkd3d_get_format(list->device, src->PlacedFootprint.Footprint.Format,
+                      true)))
+        {
+            WARN("Invalid format %#x.\n", src->PlacedFootprint.Footprint.Format);
+            return false;
+        }
+
+        vk_buffer_image_copy_from_d3d12(&out->copy.buffer_image, &src->PlacedFootprint,
+                dst->SubresourceIndex, &dst_resource->desc, out->src_format, out->dst_format, src_box, dst_x,
+                dst_y, dst_z);
+        out->copy.buffer_image.bufferOffset += src_resource->mem.offset;
+
+        out->vk_layout = d3d12_resource_pick_layout(dst_resource, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
+    }
+    else if (src->Type == D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX && dst->Type == D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX)
+    {
+        assert(d3d12_resource_is_texture(dst_resource));
+        assert(d3d12_resource_is_texture(src_resource));
+
+        out->dst_format = dst_resource->format;
+        out->src_format = src_resource->format;
+
+        vk_image_copy_from_d3d12(&out->copy.image, src->SubresourceIndex, dst->SubresourceIndex,
+                &src_resource->desc, &dst_resource->desc, out->src_format, out->dst_format,
+                src_box, dst_x, dst_y, dst_z);
+
+        /* If aspect masks do not match, we have to use fallback copies with a render pass, and there
+         * is no standard way to write to stencil without fallbacks.
+         * Checking aspect masks here is equivalent to checking formats. vkCmdCopyImage can only be
+         * used for compatible formats and depth stencil formats are only compatible with themselves. */
+        if (out->dst_format->vk_aspect_mask != out->src_format->vk_aspect_mask &&
+                (out->copy.image.dstSubresource.aspectMask & VK_IMAGE_ASPECT_STENCIL_BIT) &&
+                !list->device->vk_info.EXT_shader_stencil_export)
+        {
+            FIXME("Destination depth-stencil format %#x is not supported for STENCIL dst copy with render pass fallback.\n",
+                    out->dst_format->dxgi_format);
+            return false;
+        }
+
+        out->writes_full_subresource = d3d12_image_copy_writes_full_subresource(dst_resource,
```

Comment:
vk_layout is not set here. I think we need two. src_layout / dst_layout.

Reply:
I-to-I copies currently delegates to `d3d12_command_list_copy_image` and does not use these fields.